### PR TITLE
[EP-611, EP-614, EP-615 ] - iDeal and API base URL changes

### PR DESCRIPTION
**Pull request by gcojocaru**
- updated ideal base URL logic

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -4,3 +4,6 @@
 [submodule "TrustKit"]
 	path = TrustKit
 	url = https://github.com/datatheorem/TrustKit.git
+[submodule "CocoaDebug"]
```
Comment (stpopa) [inline .gitmodules]: please revert this

File: JudoKitObjC.xcodeproj/project.pbxproj [hunk 2]
```diff
@@ -273,6 +272,7 @@
 		82933104230AF4FB00689DC4 /* NSString+Localize.m in Sources */ = {isa = PBXBuildFile; fileRef = 82933103230AF4FB00689DC4 /* NSString+Localize.m */; };
 		8296A5E82302ED3E00E369C1 /* ApplePayButtonViewController.m in Sources */ = {isa = PBXBuildFile; fileRef = 8296A5E72302ED3E00E369C1 /* ApplePayButtonViewController.m */; };
 		DA7DF5B31C7B7A8300652A1F /* JPAddress.m in Sources */ = {isa = PBXBuildFile; fileRef = 3B9711991C761B9700E9A347 /* JPAddress.m */; };
+		F6EE9346FC31E01E87CA8134 /* Pods_JudoKitObjCExampleApp.framework in Frameworks */ = {isa = PBXBuildFile; fileRef = DDBA4165B91E4D5A429A2CF7 /* Pods_JudoKitObjCExampleApp.framework */; };
```
Comment (stpopa) [inline JudoKitObjC.xcodeproj/project.pbxproj]: please revert this and all cocoa pods stuff

File: JudoKitObjCExampleApp/AppDelegate.m [hunk 3]
```diff
@@ -33,7 +35,7 @@ @interface AppDelegate ()
 @implementation AppDelegate
 
 - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
-    
+    [CocoaDebug enable];
```
Comment (stpopa) [inline JudoKitObjCExampleApp/AppDelegate.m]: please revert this

File: JudoKitObjCExampleApp/AppDelegate.m [hunk 4]
```diff
@@ -24,6 +24,7 @@
 
 #import "AppDelegate.h"
 
+
```
Comment (mpetrenco) [inline JudoKitObjCExampleApp/AppDelegate.m]: Please run `./clang-format.sh`

File: Source/API/JPSession.m [hunk 5]
```diff
@@ -30,6 +30,7 @@
 #import "JPTransactionData.h"
 #import "JudoKit.h"
 #import "NSError+Judo.h"
+#import "JPconstants.h"
```
Comment (mpetrenco) [inline Source/API/JPSession.m]: Isn't it called `JPConstants.h` ?

File: Source/JudoKit.m [hunk 6]
```diff
@@ -546,24 +548,28 @@ - (void)invokeTokenPreAuth:(NSString *)judoId
...
-                                                                                  session:self.apiSession
-                                                                          paymentMetadata:self.paymentMetadata
-                                                                               completion:completion];
-
-    controller.modalPresentationStyle = UIModalPresentationFormSheet;
-
-    UINavigationController *navigationController = [[UINavigationController alloc] initWithRootViewController:controller];
-    self.activeViewController = controller;
-    [self.topMostViewController presentViewController:navigationController animated:YES completion:nil];
+    if ([amount.currency isEqualToString:@"EUR"]) {
```
Comment (stpopa) [inline Source/JudoKit.m]: let's extract the @"EUR" to a constant and reuse it here and in `JudoPaymentMethodsViewController.m` line 156

**Pull request merged**